💥 Major update

richi3f opened the pull request:
Introduce the following new features:

✅ Game picker (support for every game from Generation VI onwards)
✅ Color filter (filter Pokémon by color)
✅ Dependency on jQuery dropped (in favor of vanilla JS)

Temporarily disable these features:

❎ Analyze a team's weaknesses and strengths
❎ Display a copy-able team's URL

drobinson commented:
That's the weird part - I added pkmn to the team at the end of onload (after the type info is added) using the same function used for onclick, which includes the call to updateTeamTypeAnalysis().

richi3f commented:
Putting those lines at the end of the xml parsing bits (instead of at the end of onload) worked for me

drobinson commented:
Ah I was about to fix this, but I see you also fixed it!  Excited to start sharing teams with everyone.

richi3f committed:
:fire: Remove old files

richi3f committed:
:bento: Upload Pokémon renders

Upload Pokémon renders (from Pokémon HOME) for all existing Pokémon up to Calyrex and including most forms

richi3f committed:
:bento: Upload game logos

richi3f committed:
:bento: Upload Pokémon type icons

richi3f committed:
:lipstick: Add CSS

richi3f committed:
:sparkles: Update team planner

Introduce an update that completely reworks the team planner's old code.  Main new features are:

+ Game picker (support for every Gen VI or later game)
+ Color filter (Pokémon can be filtered based on their color classification)
+ Dependency on jQuery has been dropped in favor of vanilla Javascript

Some other features have been temporarily disabled or dropped:

- Search Pokémon by name
- Analyze team's weaknesses and strengths
- Display a copy-able team's URL

richi3f committed:
:page_facing_up: Update license

richi3f committed:
:memo: Update README

richi3f committed:
:bug: Prevent duplicate Pokémon from being added to team

It was possible to add multiple copies of a single Pokémon by manually adding them to the URL. This would mess up with adding/removing selectable Pokémon, so it has been fixed by implementing a check before a new Pokémon is added to the team.

richi3f committed:
:sparkles: Add search bar

Bring back the search bar to filter Pokémon by name

richi3f committed:
:sparkles: Add randomize button

Bring back the button to generate a random team

richi3f committed:
:bug: Fix visibility of empty dexes

When an empty dex (i.e., a dex that has no Pokémon because all its entries are filtered or picked), it should disappear. This was not happening when using the randomize button (e.g., picking Pokémon), but now it is fixed.
